### PR TITLE
Remove SLF4J direct dependency

### PR DESCRIPTION
**Pull request by mwoodiupui**
## References
* May help with #9321

## Description
Finish the change-over to Log4J v2 by replacing direct references to SLF4J, removing one of our direct dependencies.  SLF4J will still be present in the artifacts due to transitive dependencies, and we may need to manage its version.

## Instructions for Reviewers
List of changes in this PR:
* Replace SLF4J imports with Log4J2 imports.
* Adjust instantiation of affected loggers.
* Replace concatenated messages with parameterized templates where practical.
* Replace function-call-result parameters with function references where practical.
* Usual incidental code cleanups.

Testing:  these changes should make no difference in DSpace's operation.  Logging that is disabled may be a little faster by doing less work that is discarded.  If you see different behavior, that's a bug.  (Except for one *fixed* bug in a logging statement in the identifiers code.)

## Checklist
- [ ] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [x] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by mwoodiupui** [comment]
`dspace-server-webapp` has not been worked on yet.  That should be the last.

**Comment by mwoodiupui** [comment]
I slipped in a couple of tiny patches that are arguably relevant here:  a Checkstyle configuration fix and documentation that I had recently written for the `org.dspace.app.rest.security` package, which received a number of SLF4J edits.

**Comment by tdonohue** [comment]
Thanks for getting this ready so quickly, @mwoodiupui !  I'll put this on my list to review/test **soon** as I would like to get this merged before revisiting #9321

**Comment by mwoodiupui** [comment]
Credit is due to @tdonohue for explaining the Spring EL expressions that are implemented in o.d.a.r.security, which resulted in new package documentation once I'd studied the code a bit.

**Review by kshepherd — APPROVED** [review]
+1 thanks @mwoodiupui , changes all look good and my testing shows logging working as normal, no errors.
There are some unrelated codestyle changes introduced (eg size() checks changed to .isEmpty(), using of :: method operators, typed array usage, etc., which are all small, look fine, and I assume are intentional.

**Comment by mwoodiupui** [comment]
@kshepherd thanks for your review.  Yes, the incidental cleanups are intentional.  NetBeans calls these out in yellow (which is a little distracting) and all I have to do is click "yes, do that harmless transformation for me" so I habitually take them up.  In particular I think `isEmpty()` makes the intent clearer than `size() <= 0`.  Passing function references rather than results actually saves cycles when the log level would suppress logging so I make those changes too when I notice them.

File: dspace-server-webapp/src/main/javadoc/overview.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    The contents of this file are subject to the license and copyright
+    detailed in the LICENSE and NOTICE files at the root of the source
+    tree and available online at
+
+    http://www.dspace.org/license/
+
+-->
+<!DOCTYPE html>
+<!--
+Copyright 2024 Indiana University.
```
Comment (tdonohue) [inline dspace-server-webapp/src/main/javadoc/overview.html:12]: Can we remove this? :)

**Review by tdonohue — APPROVED** [review]
👍 @mwoodiupui : Thanks!  This looks good to me and seems to work well.  A few **minor things** though to note:
1. There's a copyright statement we should remove (see inline below)
2. There's already **more** usages of SLF4J on `main` that this PR doesn't remove.  It appears we are gaining new ones on new features.

I'm OK with merging this now as it does almost all of the cleanup.  But, it may need to have a small follow-up PR later once the features are **fully into the codebase** (two remain) to make sure none of them added SLF4J usages again.

File: dspace-server-webapp/src/main/javadoc/overview.html (hunk 1 above)
Comment (mwoodiupui) [reply dspace-server-webapp/src/main/javadoc/overview.html]: Well spotted!  Sorry, my IDE adds those automagically and I forgot to remove this one.  Done!

**Comment by mwoodiupui** [comment]
It should be noted that, because some of our direct dependencies depend on SLF4J, new direct references in DSpace code will still work...so long as those transitive dependencies remain.  But that could break if the dependency tree changes, so we should avoid SLF4J in new code.

**Comment by tdonohue** [comment]
Thanks @mwoodiupui for the updates!  Everything looks good now. Merging this immediately as it's at +2 &  seems unlikely to cause additional merge conflicts with any outstanding 8.0 features. It will also be useful for #9321 .

**Pull request merged**